pkg/client/streams.go

```diff
@@ -0,0 +1,522 @@
+	if !c.IsConnected() {
+		return nil, ErrNotConnected
+	}
+	return c.ServiceClient.StreamExecAll(ctx)
+}
+
+// StreamSet set an array of *stream.KeyValue in immudb streaming contents on a fixed size channel
+func (c *immuClient) StreamSet(ctx context.Context, kvs []*stream.KeyValue) (*schema.TxMetadata, error) {
+	if !c.IsConnected() {
+		return nil, ErrNotConnected
+	}
+
+	s, err := c.streamSet(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	kvss := c.StreamServiceFactory.NewKvStreamSender(s)
+
+	for _, kv := range kvs {
+		err = kvss.Send(kv)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return s.CloseAndRecv()
+}
+
+// StreamGet get an *schema.Entry from immudb with a stream
+func (c *immuClient) StreamGet(ctx context.Context, k *schema.KeyRequest) (*schema.Entry, error) {
+	if !c.IsConnected() {
+		return nil, ErrNotConnected
+	}
+
+	gs, err := c.streamGet(ctx, k)
+
+	kvr := c.StreamServiceFactory.NewKvStreamReceiver(gs)
+
+	key, vr, err := kvr.Next()
+	if err != nil {
+		return nil, err
+	}
+
+	value, err := stream.ReadValue(vr, c.Options.StreamChunkSize)
+	if err != nil {
+		if err == io.EOF {
+			return nil, stream.ErrMissingExpectedData
+		}
+		return nil, err
+	}
+
+	return &schema.Entry{
+		Key:   key,
+		Value: value,
+	}, nil
+}
+
+func (c *immuClient) StreamVerifiedSet(ctx context.Context, kvs []*stream.KeyValue) (*schema.TxMetadata, error) {
+	if len(kvs) == 0 {
+		return nil, errors.New("no key-values specified")
+	}
+
+	if !c.IsConnected() {
+		return nil, ErrNotConnected
+	}
+
+	err := c.StateService.CacheLock()
+	if err != nil {
+		return nil, err
+	}
+	defer c.StateService.CacheUnlock()
+
+	start := time.Now()
+	defer c.Logger.Debugf("StreamVerifiedSet finished in %s", time.Since(start))
+
+	state, err := c.StateService.GetState(ctx, c.Options.CurrentDatabase)
+	if err != nil {
+		return nil, err
+	}
+	stateTxID, err := stream.NumberToBytes(state.TxId)
+	if err != nil {
+		return nil, err
+	}
+
+	//--> collect the keys and values as they need to be used for verifications
+	stdKVs := make([]*schema.KeyValue, 0, len(kvs))
+	for i, kv := range kvs {
+		var keyBuffer bytes.Buffer
+		keyTeeReader := io.TeeReader(kv.Key.Content, &keyBuffer)
+		key := make([]byte, kv.Key.Size)
+		if _, err := keyTeeReader.Read(key); err != nil {
+			return nil, err
+		}
+		// put a new Reader back
+		kvs[i].Key.Content = bufio.NewReader(&keyBuffer)
+
+		var valueBuffer bytes.Buffer
+		valueTeeReader := io.TeeReader(kv.Value.Content, &valueBuffer)
+		value := make([]byte, kv.Value.Size)
+		if _, err = valueTeeReader.Read(value); err != nil {
+			return nil, err
+		}
+		// put a new Reader back
+		kvs[i].Value.Content = bufio.NewReader(&valueBuffer)
+
+		stdKVs = append(stdKVs, &schema.KeyValue{Key: key, Value: value})
+	}
+	//<--
+
+	s, err := c.streamVerifiableSet(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	kvss := c.StreamServiceFactory.NewKvStreamSender(s)
+
+	// 1st send the ProveSinceTx (build a "fake" KV with it):
```

Comment:
It's a bit ugly to send the `ProveSinceTx` with a fake send.
You could use directly the message sender or create a NewVeridiedKvStreamerSender